src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocUtil.java

```diff
@@ -51,6 +51,24 @@ public final class XdocUtil {
     private XdocUtil() {
     }
 
+    /**
+     * Gets all xdocs file paths. This includes Velocity .xml.vm files.
+     *
+     * @return a set of all xdocs file paths.
+     * @throws IOException if an I/O error occurs.
+     */
+    public static Set<Path> getAllXdocFilePaths() throws IOException {
```

Comment:
Why do we need this and not modify `getXdocsFilePaths`?

This will be confusing to people on when do they need to use "all" and "non-all" and how "non-all" is not everything and what is it dropping.

VM files are still XDoc files and should be included in the other to me.

Reply:
I agree. This was my initial thought, but 3 tests break if I extend `getXdocsFilePaths`. I opened https://github.com/checkstyle/checkstyle/issues/13389.
Please check it out.

Reply:
So since you moved it to a new issue, is the new issue going to be handled first?

Reply:
No, I will handle it after this gets merged.

Reply:
PR is rebased

Reply:
Removed new method and used `getXdocsFilePaths` instead